pkg/kubectl/cmd/create.go

```diff
@@ -156,9 +157,23 @@ func RunCreate(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *C
 
 // createAndRefresh creates an object from input info and refreshes info with that object
 func createAndRefresh(info *resource.Info) error {
-	obj, err := resource.NewHelper(info.Client, info.Mapping).Create(info.Namespace, true, info.Object)
-	if err != nil {
-		return err
+	var obj runtime.Object
+	var err error
+	if info.Dynamic {
+		dc, err := info.DynamicClient()
```

Comment:
The idea is that helper is largely redundant to the dynamic client, and we should eventually replace the use of helper with dynamic client.

But you are right, this is premature and doesn't belong in this PR. Trying it piecemeal like this really complicates the code. Helper should work just fine with a proper restclient for dynamic types for now. I will work on that.